### PR TITLE
HDS Login

### PR DESCRIPTION
**Pull request by voneiden**
Implement HDS login and remove old oidc-client related code. Also remove some unused example app code. 

Build doesn't currently pass as there's a bug in HDS https://helsinkisolutionoffice.atlassian.net/browse/HDS-2487

Incompatible also with earlier versions due to useGraphQL export bug, so waiting for a HDS patch release. Otherwise ready for review I suppose.

### CODE REVIEW

File: src/pages/Index.tsx [hunk 1]
```diff
@@ -1,50 +1,40 @@
...
-import { useTranslation } from 'react-i18next';
+import LoginInfo from '../components/LoginInfo';
 
 const Index = (): React.ReactElement => {
-  const clientContext = useContext(ClientContext);
   const { t } = useTranslation();
-  // No need to add the user to redux here, since all the links will redirect the user
-  // to a new page, thus refreshing the store. The user is re-fetched and added to redux in FormStepper.tsx
-  const userProfile = useUserProfile() as UserProfile | GraphQLClientError;
+  useApiTokens();
```
Comment (mikkojamG) [inline src/pages/Index.tsx:17]: This does not seem to be used for anything 🤔

**Review by mikkojamG — CHANGES_REQUESTED** [review]

File: src/components/Header.tsx [hunk 2]
```diff
@@ -8,28 +8,83 @@ import {
...
 
+const Header = (): React.ReactElement => {
   const { t } = useTranslation();
 
   const title = t('common:title');
 
   document.title = title;
-
-  const userName = user ? `${user.given_name} ${user.family_name}` : '';
+  useOidcClientTracking();
```
[the remark this answers is not in the document]
Comment (voneiden) [reply src/components/Header.tsx:86]: This one's actually really important for ensuring that  `WithAuthentication` gets re-rendered when user changes. `useOidcClientTracking()` forces component re-rendering when user is 

- USER_UPDATED
- USER_REMOVED
- USER_RENEWAL_STARTED

But I will still revisit the usage as there's the second WithAuthentication.

File: src/App.tsx [hunk 3]
```diff
@@ -1,65 +1,51 @@
...
+  isApiTokensRemovedSignal,
+  isApiTokensUpdatedSignal,
+  SessionEndedHandler,
+  useSignalListener
+} from 'hds-react';
 import { useTranslation } from 'react-i18next';
-import { selectPromptLogin, setPromptLogin } from './components/user/userSlice';
+import LoginCallbackHandlerWrapper from './components/LoginCallbackHandlerWrapper';
 import RequestLoader from './components/loader/RequestLoader';
+import { AuthError } from './pages/authError';
```
[the remark this answers is not in the document]
Comment (voneiden) [reply src/App.tsx:15]: It seems I had a few small commits I failed to push :sob:

File: src/components/Header.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (voneiden) [reply src/components/Header.tsx:84]: useOidcClientTracking(); is now used in three places. There was also the <ProtectedRoute> that I missed earlier.

**Comment by voneiden** [comment]
I originally thought I need  `useApiTokens` with `useGraphQL`, but it seems everything works now fine with just OidcClientTracking.

File: src/components/PageContainer.tsx [hunk 4]
```diff
@@ -1,15 +1,10 @@
 import React from 'react';
 
 import styles from './styles.module.css';
-import ErrorPrompt from './ErrorPrompt';
+//import ErrorPrompt from './ErrorPrompt';
```
Comment (mikkojamG) [inline src/components/PageContainer.tsx:4]: Unused import here 👀

**Review by mikkojamG — CHANGES_REQUESTED** [review]

**Review by mikkojamG — APPROVED** [review]
LGTM (besides FileInput bug) 👀

**Comment by terovirtanen** [comment]
**PYSAKOINNIN-SAHK-ASIOINTI-UI branch is deployed to platta: https://pysakoinnin-sahk-asiointi-pr169-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://pysakoinnin-sahk-asiointi-pr169-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**